ed_sensor_integration/src/laser/updater.cpp

```diff
@@ -0,0 +1,655 @@
+
+    for (unsigned int i_cluster = 0; i_cluster < clusters.size(); ++i_cluster)
+    {
+        const EntityUpdate& cluster = clusters[i_cluster];
+
+        // Get the assignment for this cluster
+        int i_entity = assig[i_cluster];
+
+        ed::UUID id;
+        ed::ConvexHull new_chull;
+        geo::Pose3D new_pose;
+
+        if (i_entity == -1)
+        {
+            // No assignment, so add as new cluster
+            new_chull = cluster.chull;
+            new_pose = cluster.pose;
+
+            // Generate unique ID
+            id = ed::Entity::generateID().str() + "-laser";
+
+            // Update existence probability
+            req.setExistenceProbability(id, 1.0); // TODO magic number
+        }
+        else
+        {
+            // Mark the entity as being associated
+            //entities_associated[i_entity] = i_cluster;
+
+            // Update the entity
+            const ed::EntityConstPtr& e = entities[i_entity];
+
+            if (!e->hasFlag("locked"))
+            {
+                new_chull = cluster.chull;
+                new_pose = cluster.pose;
+            }
+
+            // Update existence probability
+            double p_exist = e->existenceProbability();
+            req.setExistenceProbability(e->id(), std::min(1.0, p_exist + 0.1)); // TODO: very ugly prob update
+
+            id = e->id();
+        }
+
+        // Set convex hull and pose
+        if (!new_chull.points.empty())
+        {
+            req.setConvexHullNew(id, new_chull, new_pose, timestamp, lrf_frame_);
+
+            // --------------------------
+            // Temp for RoboCup 2015; todo: remove after
+
+            if (!cluster.flag.empty())
+                req.setFlag(id, cluster.flag);
+
+            // --------------------------
+        }
+
+        // Set timestamp
+        req.setLastUpdateTimestamp(id, timestamp);
+    }
+
+    // - - - - - - - - - - - - - - - - - -
+    // Clear unassociated entities in view
+
+    //    for(unsigned int i = 0; i < entities_associated.size(); ++i)
+    //    {
+    //        const ed::EntityConstPtr& e = entities[i];
+
+    //        // If the entity is associated, skip it
+    //        if (entities_associated[i] >= 0)
+    //            continue;
+
+    //        const geo::Pose3D& pose = e->pose();
+
+    //        // Transform to sensor frame
+    //        geo::Vector3 p = sensor_pose.inverse() * pose.t;
+
+    //        if (!pointIsPresent(p, lrf_model_, sensor_ranges))
+    //        {
+    //            double p_exist = e->existenceProbability();
+    //            if (p_exist < 0.3) // TODO: magic number
+    //                req.removeEntity(e->id());
+    //            else
+    //            {
+    //                req.setExistenceProbability(e->id(), std::max(0.0, p_exist - 0.1));  // TODO: very ugly prob update
+    //            }
+    //        }
+    //    }
+
+    //std::cout << "Total took " << t_total.getElapsedTimeInMilliSec() << " ms." << std::endl;
```

Comment:
Determine what to do with this code.

Reply:
We keep it for now. It is good functionality but it requires more work to integrate it. Therefore I would leave it like this for someone to stumble upon in the future

Reply:
Maybe add that as a comment to the code.